compiler-core/src/parse.rs

```diff
@@ -2167,14 +2167,27 @@ where
                         self.parse_const_record_finish(start, Some(name), upname, end)
                     }
                     Some((_, Token::Name { name: end_name }, end)) => {
-                        let _ = self.next_tok(); // upname
-                        Ok(Some(Constant::Var {
-                            location: SrcSpan { start, end },
-                            module: Some(name),
-                            name: end_name,
-                            constructor: None,
-                            typ: (),
-                        }))
+                        if self.peek_tok1() == Some(&Token::LeftParen) {
+                            let _ = self.next_tok(); // name
+
+                            parse_error(
+                                ParseErrorType::UnexpectedFunction,
+                                SrcSpan {
+                                    start,
+                                    end: end + 1,
+                                },
+                            )
+                        } else {
+                            let _ = self.next_tok(); // name
```

Comment:
This is the same in both branches so let's lift it to before the `if`

Reply:
I initially did this because if I understand correctly, `next_tok` pops the current token, which would made `peek_tok1` return another token and not the one that could be a LeftParen. I've changed that to instead pop the token and then just match `self.tok0`. https://github.com/gleam-lang/gleam/pull/1839/commits/0982dc4db6eacf6609a1e64cef21d5c0c20f584a

Happy to change this if there's a better way to do it.